### ISSUE TITLE
Enviroplus install not working on bookworm / Debian 12

### ISSUE THREAD

**Issue by p8mty** [pimoroni/get#218]
There seems to be a lot of issues installing the Enviro+ support under bookworm, some missing Python packages, but mainly to do with things around VENV.  The install works fine on bullseye.

I've only tried Enviroplus, but I presume other boards are the same.

If nothing else, then the documentation might be updated to say that it currently is not working on bookworm.

**Comment by Trotter73** [comment]
There is a bit of a mess around Bookworm and Python at the moment, I understand the guys here are working hard to get that resolved in a stable manor... 

You don't say what Pi you are using, but I would suggest if you are not using a Pi5, to use the Legacy image for a while until these guys have had time to catch up...

**Comment by p8mty** [comment]
I was using a ZeroW for that particular task.  
I understand it's all to do with the new world order of Python and virtual environments.  But I've seen comments elsewhere along the lines of "one star - it's all broken.", or "Don't bother buying the Enviro+, none of the software works". I just think that a stop gap measure of putting a note on the Pimoroni web site of the affected devices that Bookworm support is in the pipeline and to use Legacy / bullseye in the mean time if possible.

**Comment by Trotter73** [comment]
I have about 1/2 dozen python projects/toys running, they are all broken in some way with Bookworm, either Python, GPIO or audio, I'm staying on Legacy for the foreseeable future.

Is this what you have got [https://shop.pimoroni.com/products/enviro?variant=31155658489939] ? If so this is quite a cool project, learnt a lot from it... [https://github.com/nophead/EnviroPlusWeb]

**Comment by p8mty** [comment]
Yes, that's what I have.  I use it to feed data as an IoT device to Home Assistant.  But I thought I would update it, improving the screen content, that sort of thing, so I updated to the latest OS. Instantly regretted it. It's on legacy now and much happier.

**Comment by Gadgetoid** [comment]
For anyone who's interested in the wider scope of the problem, here's my Pi 5 - 🔥 Aaah Everything Is On Fire 🔥  TODO list and some details of what's in progress/updated/broken that might be useful to know - https://github.com/pimoroni/boilerplate-python/issues/16

I need to tweak the "get" scripts to be simpler- ie: just to grab and run of the installer in the git repo, but that wont happen until the fixes are done/merged.

For Enviro Plus the updated installer, libraries and such are a WIP here - https://github.com/pimoroni/enviroplus-python/pull/126

Biggest blocker at the moment is that I'm trying to update things without really being sure of how things could and should work in the new Pi 5, post RPi.GPIO world. GPIO lines don't seem to have the good sense to be named the same or found in the same place across Pi versions 😬

**Comment by helgibbons** [comment]
I think we've now updated all the references to the outdated get installer for Enviro+ but if anybody spots any that we've missed, please let us know.
